internal/mode/static/telemetry/collector_test.go

```diff
@@ -303,62 +340,334 @@ var _ = Describe("Collector", Ordered, func() {
 		})
 	})
 
-	Describe("clusterID collector", func() {
-		When("collecting clusterID", func() {
+	Describe("cluster information collector", func() {
+		When("collecting node count data", func() {
+			It("collects correct data for one node", func() {
+				k8sClientReader.ListCalls(createListCallsFunc(nodeList))
+
+				expData.Data.ClusterNodeCount = 1
+
+				data, err := dataCollector.Collect(ctx)
+
+				Expect(err).To(BeNil())
+				Expect(expData).To(Equal(data))
+			})
+
 			When("it encounters an error while collecting data", func() {
-				It("should error if the kubernetes client errored when getting the namespace", func() {
-					expectedError := errors.New("there was an error getting clusterID")
-					k8sClientReader.GetCalls(mergeGetCallsWithBase(
-						func(_ context.Context, _ types.NamespacedName, object client.Object, _ ...client.GetOption) error {
-							switch object.(type) {
-							case *v1.Namespace:
-								return expectedError
-							}
-							return nil
-						}))
+				It("should error when there are no nodes", func() {
+					expectedError := errors.New("failed to collect cluster information: NodeList length is zero")
+					k8sClientReader.ListCalls(createListCallsFunc(nil))
+
+					_, err := dataCollector.Collect(ctx)
+
+					Expect(err).To(MatchError(expectedError))
+				})
+				It("should error on kubernetes client api errors", func() {
+					expectedError := errors.New("there was an error getting NodeList")
+					k8sClientReader.ListReturns(expectedError)
 
 					_, err := dataCollector.Collect(ctx)
 					Expect(err).To(MatchError(expectedError))
 				})
 			})
 		})
-	})
 
-	Describe("node count collector", func() {
-		When("collecting node count data", func() {
-			It("collects correct data for no nodes", func() {
-				k8sClientReader.ListCalls(createListCallsFunc(nil))
+		When("collecting cluster platform data", func() {
```

Comment:
better to move the tests that collect a platform to a unit for getPlatform function.
That function is well isolated and it's gonna be much easier if we just those cases there.